packages/lesswrong/components/hooks/useAuth0Client.ts

```diff
@@ -0,0 +1,75 @@
+import { auth0ClientSettings } from "../../lib/publicSettings";
+import { combineUrls } from "../../lib/vulcan-lib";
+
+class Auth0ClientError extends Error {
+  constructor(message?: string | null) {
+    super(message || "Something went wrong");
+  }
+}
+
+class Auth0Client {
+  private async post(endpoint: string, body: JsonRecord): Promise<void> {
+    const result = await fetch(endpoint, {
+      method: "POST",
+      body: JSON.stringify(body),
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+    if (result.status !== 200) {
+      const data = await result.json();
+      let message = data.error || data.error_description || "Something went wrong";
+      // Remove the ugly full-stop that auth0 appends to its error messages
+      if (message[message.length - 1] === ".") {
+        message = message.slice(0, -1);
+      }
+      throw new Auth0ClientError(message);
+    }
+  }
+
+  async login(email: string, password: string): Promise<void> {
```

Comment:
Instead of dealing with Auth0's mess of third-party cookies, we now handle login and signup with username/password by sending the credentials to our own server and handling the authentication there. The endpoints are registered in `authenticationMiddlewares.ts` and the implementations are in `authentication/auth0.ts`

Reply:
@Will-Howard Do you know if we need to do anything (/can do anything) to prevent datadog from storing the bodies of these requests?